crystal-map-connector/src/main/kotlin/ch/ergon/dope/extension/type/objectfunction/ObjectFunction.kt

```diff
@@ -0,0 +1,94 @@
+package ch.ergon.dope.extension.type.objectfunction
+
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.addAttribute
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.concat
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.innerPairs
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.length
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.names
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.pairs
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.pairsNested
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.paths
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.putAttribute
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.removeAttribute
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.renameAttribute
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.replace
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.unwrap
+import ch.ergon.dope.resolvable.expression.unaliased.type.function.`object`.values
+import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
+import ch.ergon.dope.toDopeType
+import ch.ergon.dope.validtype.ObjectType
+import ch.ergon.dope.validtype.StringType
+import ch.ergon.dope.validtype.ValidType
+import com.schwarz.crystalapi.schema.CMObjectField
+import com.schwarz.crystalapi.schema.CMType
+import com.schwarz.crystalapi.schema.Schema
+
+fun CMObjectField<Schema>.addAttribute(key: TypeExpression<StringType>, value: TypeExpression<out ValidType>) =
+    toDopeType().addAttribute(key, value)
+
+fun CMObjectField<Schema>.addAttribute(key: String, value: TypeExpression<out ValidType>) =
+    toDopeType().addAttribute(key.toDopeType(), value)
+
+fun CMObjectField<Schema>.addAttribute(key: TypeExpression<StringType>, value: CMType) = toDopeType().addAttribute(key, value.toDopeType())
+
+fun CMObjectField<Schema>.addAttribute(key: String, value: CMType) = toDopeType().addAttribute(key.toDopeType(), value.toDopeType())
+
+fun TypeExpression<ObjectType>.addAttribute(key: TypeExpression<StringType>, value: CMType) = addAttribute(key, value.toDopeType())
```

Comment:
Couldn't the key also be a CM string field?

Reply:
Yes its possible.
done